Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc

```diff
@@ -0,0 +1,146 @@
+#include "CondFormats/DataRecord/interface/EcalChannelStatusRcd.h"
+#include "CommonTools/Utils/interface/StringToEnumValue.h"
+#include <iostream>
+#include <sstream>
+
+ECALpedestalPCLHarvester::ECALpedestalPCLHarvester(const edm::ParameterSet& ps):
+    currentPedestals_(0),channelStatus_(0){
+
+    chStatusToExclude_= StringToEnumValue<EcalChannelStatusCode::Code>(ps.getParameter<std::vector<std::string> >("ChannelStatusToExclude"));
+    minEntries_=ps.getParameter<int>("MinEntries");
+
+}
+
+void ECALpedestalPCLHarvester::dqmEndJob(DQMStore::IBooker& ibooker_, DQMStore::IGetter& igetter_) {
+    
+
+    // calculate pedestals and fill db record
+    EcalPedestals pedestals;
+
+    std::cout << "Filling ped record " << std::endl;
+       
+    for (uint16_t i =0; i< EBDetId::kSizeForDenseIndexing; ++i) {
+        std::stringstream hname;
+        hname<<"EcalCalibration/EcalPedestalPCL/";
+        hname<<"eb_"<<i;
+        MonitorElement* ch= igetter_.get(hname.str());
+        double mean = ch->getMean();
+        double rms  = ch->getRMS();
+
+       DetId id = EBDetId::detIdFromDenseIndex(i);
+        EcalPedestal ped;
+        EcalPedestal oldped=* currentPedestals_->find(id.rawId());
+
+        ped.mean_x12=mean;
+        ped.rms_x12=rms;
+
+        // if bad channel or low stat skip
+        if(ch->getEntries()< minEntries_ || !checkStatusCode(id)){
+            
+            ped.mean_x12=oldped.mean_x12;
+            ped.rms_x12=oldped.rms_x12;
+            
+        }
+       
+        ped.mean_x6=oldped.mean_x6;
+        ped.rms_x6=oldped.rms_x6;
+        ped.mean_x1=oldped.mean_x1;
+        ped.rms_x1=oldped.rms_x1;
+
+        pedestals.setValue(id.rawId(),ped);
+    }
+
+    for (uint16_t i =0; i< EEDetId::kSizeForDenseIndexing; ++i) {
+        std::stringstream hname;
+        hname<<"EcalCalibration/EcalPedestalPCL/";
+        hname<<"ee_"<<i;
+        MonitorElement* ch= igetter_.get(hname.str());
+        double mean = ch->getMean();
+        double rms  = ch->getRMS();
+
+        DetId id = EEDetId::detIdFromDenseIndex(i);
+        EcalPedestal ped;
+        EcalPedestal oldped= *currentPedestals_->find(id.rawId());
+
+        ped.mean_x12=mean;
+        ped.rms_x12=rms;
+
+        // if bad channel or low stat skip
+        if(ch->getEntries()< minEntries_ || !checkStatusCode(id)){
+            ped.mean_x12=oldped.mean_x12;
+            ped.rms_x12=oldped.rms_x12;
+        }
+
+        ped.mean_x6=oldped.mean_x6;
+        ped.rms_x6=oldped.rms_x6;
+        ped.mean_x1=oldped.mean_x1;
+        ped.rms_x1=oldped.rms_x1;
+
+        pedestals.setValue(id.rawId(),ped);
+    }
+
+
+    std::cout << "Writing to DB file ... " << std::endl;
+
+    // write out pedestal record
+    edm::Service<cond::service::PoolDBOutputService> poolDbService;
+
+    if( poolDbService.isAvailable() )
+        poolDbService->writeOne( &pedestals, poolDbService->currentTime(),
+                                 "EcalPedestalsRcd"  );
+    else
```

Comment:
I wasn't sure at first,

this: 
     https://github.com/cms-sw/cmssw/blob/CMSSW_9_1_X/CondCore/DBOutputService/interface/PoolDBOutputService.h#L139

      // Service time utility method 
      // return the time value of the current edm::Event according to the 
      // given timetype

is indeed the time we want to set this IOV since boundary